Add Spotless plugin (#1435)
jenkinsci/gitlab-plugin: CONTRIBUTING.md

* you updated the help docs
* you updated the README
* you have used spotbugs to see if you haven't introduced any new warnings

> * you have run `mvn spotless:apply` for confirming that the code formatting is correct

Reviewer: ```suggestion
* you can run `mvn spotless:apply` to confirm that the code formatting is as expected
```

Reviewer: Can we have spotless bound to the maven phase as they describe [here](https://github.com/diffplug/spotless/blob/main/plugin-maven/README.md#binding-to-maven-phase)? I am stating this here just so we are not confused. I have checked that this is already the case and we do not need to run this in addition to `mvn verify` to have the spotless check included.
